prime/src/pages/BorrowActionsPage.tsx

```diff
@@ -465,21 +463,19 @@ export default function BorrowActionsPage() {
     }
     const { token0, token1 } = marginAccount;
     const earnedFeesValues = Object.values(uniswapPositionEarnedFees);
-    const priceX96 = marginAccount.sqrtPriceX96.mul(marginAccount.sqrtPriceX96).div(BIGQ96);
-    const token0FeesEarned = new Big(earnedFeesValues.reduce((p, c) => p.add(c.token0FeesEarned), new Big(0))).mul(
-      10 ** token0.decimals
-    );
-    const token1FeesEarned = new Big(earnedFeesValues.reduce((p, c) => p.add(c.token1FeesEarned), new Big(0))).mul(
-      10 ** token1.decimals
-    );
-    const token0FeesEarnedInTermsOfToken1 = priceX96.mul(token0FeesEarned).div(BIGQ96);
+    const priceX96 = marginAccount.sqrtPriceX96.square();
+    const token0FeesEarned = earnedFeesValues.reduce((p, c) => p.add(c.token0FeesEarned), GN.zero(token0.decimals));
+    const token1FeesEarned = earnedFeesValues.reduce((p, c) => p.add(c.token1FeesEarned), GN.zero(token1.decimals));
+    const token0FeesEarnedInTermsOfToken1 = priceX96.mul(token0FeesEarned).setResolution(token1.decimals);
```

Comment:
In order for the resolution stuff to work out correctly, you need to change the terms' order here. Should be `token0FeesEarned.mul(priceX96).setResolution(token1.decimals)`